### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by mimansha-swarup**
- add to cart
- remove from cart
- change quantity

- add to wishist
- remove from wishList
- move to bag

https://user-images.githubusercontent.com/69811810/161190663-8473cea4-577d-4ca4-959a-0409733fd85b.mp4

ps:  haven't done with private n public route yet, so do login before trying

### CODE REVIEW

**Review by bipin7yadav — APPROVED** [review]
ui looks great

File: src/Context/cartContext.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            type: type,
+          },
+        },
+        {
+          headers: {
+            authorization: token,
+          },
+        }
+      );
+      console.log(response)
```
Comment (srishti-maurya) [inline src/Context/cartContext.js]: you may remove `console.log`

File: src/Context/cartContext.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      if (response.status === 200) {
+        cartDispatch({ type: cartActions[type.toUpperCase()],payload:product });
+      }
+    } catch (error) {
+      
+    }
+  };
```
Comment (srishti-maurya) [inline src/Context/cartContext.js:87]: I assume that error handling would be handled later and here it is empty intentionally

File: src/Page/Cart/CartPage.js [hunk 3]
```diff
@@ -1,19 +1,43 @@
...
         <div className="action">
           <button className="btn btn-contained teal">PLACE ORDER</button>
         </div>
       </div>
       <CartDetails />
+        
+          </>
+      ):(
+        <div className="flex flex-column center mx-auto mt-3">
+          <img class="width-half mb-3 wishList-img" src={shopNowSVG} alt="empty bag"/>
```
Comment (srishti-maurya) [inline src/Page/Cart/CartPage.js:29]: you may give `alt` attribute as `product.title` maybe

**Review by srishti-maurya — APPROVED** [review]
Code looks clean and readable.
UI is awesome.
Amazing work

**Review by srishti-maurya — COMMENTED** [review]
Suggestion: When a user is not logged in, in that case you may redirect the user to login/ signup page if user goes to cart or wishlist page. i.e. you may make cart and wishlist a private route

File: src/Context/cartContext.js (hunk 1 above)
Comment (mimansha-swarup) [reply src/Context/cartContext.js]: yeah thanks

File: src/Context/cartContext.js (hunk 2 above)
Comment (mimansha-swarup) [reply src/Context/cartContext.js:87]: yesd

**Pull request merged**